### PR TITLE
Add keyboard jisplit89

### PR DESCRIPTION
**Pull request by Salicylic-acid3**
Add jisplit89 keyboard.

A 89 keys JIS Layout keyboard.
Salicylic-acid3

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/jisplit89/keymaps/default/keymap.c [hunk 1]
```diff
@@ -0,0 +1,83 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/jisplit89/keymaps/default/keymap.c]: This file shouldn't be needed.  All the information that you'd need from this has functions for getting that info. 

```suggestion
```

File: keyboards/jisplit89/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+int RGB_current_mode;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
+  switch (keycode) {
+    #ifdef RGBLIGHT_ENABLE
+      case RGB_MOD:
+          if (record->event.pressed) {
+            rgblight_mode(RGB_current_mode);
+            rgblight_step();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/jisplit89/keymaps/default/keymap.c]:
```suggestion
            RGB_current_mode = rgblight_get_mode();
```

File: keyboards/jisplit89/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+          }
+        break;
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/jisplit89/keymaps/default/keymap.c]:
```suggestion
            RGB_current_mode = rgblight_get_mode();
```

File: keyboards/jisplit89/keymaps/salicylic/keymap.c [hunk 4]
```diff
@@ -0,0 +1,151 @@
+#include QMK_KEYBOARD_H
+#include "keymap_jp.h"
+
+#ifdef RGBLIGHT_ENABLE
+//Following line allows macro to read current RGB settings
+extern rgblight_config_t rgblight_config;
+#endif
```
Comment (drashna) [inline keyboards/jisplit89/keymaps/salicylic/keymap.c]:
```suggestion
```

File: keyboards/jisplit89/keymaps/salicylic/keymap.c [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+int RGB_current_mode;
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  bool result = false;
+  switch (keycode) {
+    #ifdef RGBLIGHT_ENABLE
+      case RGB_MOD:
+          if (record->event.pressed) {
+            rgblight_mode(RGB_current_mode);
+            rgblight_step();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/jisplit89/keymaps/salicylic/keymap.c]:
```suggestion
            RGB_current_mode = rgblight_get_mode();
```

File: keyboards/jisplit89/keymaps/salicylic/keymap.c [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+          }
+        break;
+      case RGB_RST:
+          if (record->event.pressed) {
+            eeconfig_update_rgblight_default();
+            rgblight_enable();
+            RGB_current_mode = rgblight_config.mode;
```
Comment (drashna) [inline keyboards/jisplit89/keymaps/salicylic/keymap.c]:
```suggestion
            RGB_current_mode = rgblight_get_mode();
```

File: keyboards/jisplit89/readme.md [hunk 7]
```diff
@@ -0,0 +1,17 @@
+# 7skb
```
Comment (mtei) [inline keyboards/jisplit89/readme.md]: Is the name of this keyboard 7sKB or jisplit89?

File: keyboards/jisplit89/readme.md [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+* Hardware Supported: jisplit89 PCB, Pro Micro
+* Hardware Availability: [PCB & Case Data](https://github.com/Salicylic-acid3/PCB_Data), [Booth Shop](https://salicylic-acid3.booth.pm/items/1916810)
+
+Make example for this keyboard (after setting up your build environment):
+
+    make jisplit89:default
+
+See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
+
+[Build guide](https://salicylic-acid3.hatenablog.com/entry/7skb-mx-build-guide)
```
Comment (mtei) [inline keyboards/jisplit89/readme.md]: This link looks like a link to the 7sKB build guide.
Please add a description.

File: keyboards/jisplit89/rev1/config.h [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0xFEED
+#define PRODUCT_ID      0x3060
+#define DEVICE_VER      0x0009
```
Comment (hsgw) [inline keyboards/jisplit89/rev1/config.h]: I think you should use unique VID/PID pair.
It is wrong to use Device Version to identify the device.

File: keyboards/jisplit89/rev1/rev1.h [hunk 10]
```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "jisplit89.h"
+
+#include "quantum.h"
+
+//////////////////////////////////////////////////////////////////////////////
+// When only use 7skb.
```
Comment (shelaf) [inline keyboards/jisplit89/rev1/rev1.h]:
```suggestion
// When only use JISplit89.
```

File: keyboards/jisplit89/jisplit89.h [hunk 11]
```diff
@@ -0,0 +1,7 @@
+#pragma once
+
+#include "quantum.h"
+
+#ifdef KEYBOARD_jisplit89_rev1
```
Comment (shelaf) [inline keyboards/jisplit89/jisplit89.h]: This definition is not useful.

File: keyboards/jisplit89/readme.md (hunk 8 above)
Comment (Salicylic-acid3) [reply keyboards/jisplit89/readme.md]: Thanks
Annotation was added.

File: keyboards/jisplit89/rev1/config.h (hunk 9 above)
Comment (Salicylic-acid3) [reply keyboards/jisplit89/rev1/config.h]: Thanks
PID changed.

File: keyboards/jisplit89/jisplit89.h (hunk 11 above)
Comment (Salicylic-acid3) [reply keyboards/jisplit89/jisplit89.h]: Thanks
Remove unnecessary definitions.

**Comment by Salicylic-acid3** [comment]
What should I do next?

File: keyboards/jisplit89/keymaps/default/config.h [hunk 12]
```diff
@@ -0,0 +1,22 @@
+/* Copyright 2020 Salicylic_acid3
```
Comment (zvecr) [inline keyboards/jisplit89/keymaps/default/config.h]: I think the defaults should be used here, which means we might as well delete this file.

File: keyboards/jisplit89/rules.mk [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = caterina
+
+# Build Options
+#   change to "no" to disable the options, or define them in the Makefile in
+#   the appropriate keymap folder that will get included automatically
```
Comment (noroadsleft) [inline keyboards/jisplit89/rules.mk]:
```suggestion
#   change yes to no to disable

```

File: keyboards/jisplit89/rules.mk [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+NKRO_ENABLE = no            # Nkey Rollover - if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = yes       # Enable WS2812 RGB underlight. 
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no    # Breathing sleep LED during USB suspend
```
Comment (noroadsleft) [inline keyboards/jisplit89/rules.mk]: Updates to match QMK's AVR template.

```suggestion
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
MIDI_ENABLE = no            # MIDI support
BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
AUDIO_ENABLE = no           # Audio output on port C6
FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
UNICODE_ENABLE = no         # Unicode
```

Also I'd prefer if all these contents were in `keyboards/jisplit89/rev1/rules.mk` instead.

**Comment by Salicylic-acid3** [comment]
Thanks for your comment.
What should I do next?

**Comment by drashna** [comment]
Thanks!

**Pull request merged**